aieng-ui/backend/app/runtime_registry/cae.py

```
@@ -2301,16 +2437,26 @@ def _with_run_solver_receipt(result: dict[str, Any]) -> dict[str, Any]:
-            def _write_safe(artifact_path: str, source: _Path) -> None:
+            def _write_safe(artifact_path: str, source: _Path) -> bool:
+                """Copy `source` into the package; report whether it landed.
+
+                The return value is load-bearing for the FRD: with
+                `overwrite=False` and a result already sitting under this run,
+                the package keeps the OLD file while this run's metrics come
+                from the new one. A caller that assumes the write happened
+                names the old member as the new metrics' source.
+                """
                 try:
                     art = write_artifact_to_package(
```

Ruff (0.16.3)

[error] Undefined name `write_artifact_to_package`

(F821)